llvm/lib/Transforms/Scalar/MemCpyOptimizer.cpp

```diff
@@ -158,11 +158,16 @@ class MemsetRanges {
 
   /// A sorted list of the memset ranges.
   SmallVector<MemsetRange, 8> Ranges;
-
-  const DataLayout &DL;
+  const DataLayout *DL;
 
 public:
-  MemsetRanges(const DataLayout &DL) : DL(DL) {}
+  MemsetRanges() {}
```

Comment:
Is this because MapVector requires a default ctor?

Reply:
Yes, it requires.